src/libraries/Microsoft.PowerFx.Interpreter/Functions/IRandomService.cs

```diff
@@ -38,5 +47,28 @@ public virtual double NextDouble()
                 return _random.NextDouble();
             }
         }
+
+        public virtual decimal NextDecimal()
+        {
+            lock (_randomizerLock)
+            {
+                var randChars = new char[30];
+                randChars[0] = '0';
+                randChars[1] = '.';
+
+                for (int i = 2; i < 30; i++)
+                {
+                    randChars[i] = (char)_random.Next(48, 58);
+                }
+
+                var randStr = new string(randChars);
+                if (decimal.TryParse(randStr, out decimal decimalResult))
```

Comment:
>randStr,  [](http://example.com/codeflow?start=37&length=9)

Pass invariant culture, since we are using `.` as the decimal separator

Reply:
passed invariant culture here.